tls/s2n_ktls_io.c

```diff
@@ -325,6 +367,9 @@ ssize_t s2n_ktls_sendv_with_offset(struct s2n_connection *conn, const struct iov
 
 int s2n_ktls_send_cb(void *io_context, const uint8_t *buf, uint32_t len)
 {
+    POSIX_ENSURE_REF(io_context);
+    POSIX_ENSURE_REF(buf);
```

Comment:
I added this to handle a failure I saw in the tests after rebasing. This change matches our existing non-ktls implementation of the send callback: https://github.com/aws/s2n-tls/blob/bb6052e265317895aab4cd36edff1a558225ab15/utils/s2n_socket.c#L209-L212 Considering NULL an error makes sense, since the buf is always a reference to known non-null memory in the conn->out buffer.

The root cause of the test failures was https://github.com/aws/s2n-tls/issues/4191 Before I added the tests for this PR, the last sendmsg callback set before the s2n_ktls_send_cb safety tests was s2n_test_ktls_sendmsg_fail. When giving a zeroed context, s2n_test_ktls_sendmsg_fail causes s2n_sendmsg to fail with an errno of 0, which we interpret as S2N_ERR_IO. After I added the tests for this PR, the last sendmsg callback set was the testutils stuffer io callback, which fails with S2N_ERR_IO_BLOCKED when given a zeroed context. So the behavior changed.

I could have fixed this by explicitly setting a callback with known handling of NULLs, but it seemed more correct to just add the NULL checks to s2n_ktls_send_cb.